### PR TITLE
refactor(qgrads): refactored interface for autograd and mathoperations

### PR DESCRIPTION
**Pull request by LeoBuron**
The interface allows any quantization function for forward and backward. This gives  more freedom for quantization in the future.. 

BREAKING CHANGE: This is a breaking change of the mathops and autograd interface in quantized grads

### CODE REVIEW

File: elasticai/creator/nn/quantized_grads/base_modules/sigmoid.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import torch
 from torch.nn import Sigmoid as _Sigmoid
 
-from elasticai.creator.nn.quantized_grads._math_operations import MathOperations
+from .. import MathOperations
```
Comment (glencoe) [inline elasticai/creator/nn/quantized_grads/base_modules/sigmoid.py:4]: Not referring to a leaf module here seems dangerous to me. The problem is: to load the parent module, python will typically have to load child modules, if someone decides to add add a symbol from `sigmoid` to the `base_modules` module, we get a circular import. Therefore we should keep symbols exported by `__init__.py` files for external clients and not import them in child modules.

**Review by glencoe — CHANGES_REQUESTED** [review]
Except for that one import and the actual commit message, this one looks good to me.

Wrt. the commit message: Please summarize what you wrote in the corresponding issue and in the PR description in the long text of the commit message.

What is the breaking change you're referring to exactly? If your changes include a breaking change, this cannot be a refactor commit, better call it a feature then. I mean you're introducing the change to improve your workflow, right? So i'd argue it's a feature or part of a feature. Please ensure that the breaking change description makes very clear, that it's only breaking the mathops in the qgrads package and not somewhere else, otherwise you might have people worried :D.

**Comment by glencoe** [comment]
Oh... and before i forget: @julianhoever  and me had a discussion about the location of test files at #442. You might wanna have a quick look there.

**Comment by glencoe** [comment]
good test coverage btw ;) 👍
